packages/transaction-controller/src/TransactionController.ts

```diff
@@ -438,14 +451,34 @@ export class TransactionController extends BaseController<
       transactions: [],
     };
     this.initialize();
-    const provider = getProvider();
+    this.provider = provider;
     this.getNetworkState = getNetworkState;
     this.ethQuery = new EthQuery(provider);
     this.registry = new MethodRegistry({ provider });
+    this.nonceTracker = new NonceTracker({
+      provider,
+      blockTracker,
+      getPendingTransactions: (address) => {
+        return formatTxForNonceTracker(
+          this.state.transactions.filter(
+            ({ status, transaction: { from } }) =>
+              status === TransactionStatus.submitted &&
+              (address ? from.toLowerCase() === address.toLowerCase() : true),
```

Comment:
Hmm, I'd expect the address to always be set. Is there a case where it's missing?

Reply:
you're correct. `getNonceLock` where these methods are called does not have the address marked as optional. Not sure why I thought we needed this check.

Reply:
addressed here: https://github.com/MetaMask/core/pull/1147/commits/2138d654c19b87a7e9baf2221fad057ed3434975